activerecord/lib/active_record/migration.rb

```diff
@@ -133,6 +133,9 @@ class PendingMigrationError < MigrationError #:nodoc:
 
     action "Run pending migrations" do
       ActiveRecord::Tasks::DatabaseTasks.migrate
+      if ActiveRecord::Base.dump_schema_after_migration
+        ActiveRecord::Tasks::DatabaseTasks.dump_schema(ActiveRecord::Tasks::DatabaseTasks.current_config)
+      end
```

Comment:
Huh, so this and the previous code will only work for single database applications. Is that the intention of this `action` code? Or are we just getting these working before we consider multi-db apps?

Reply:
I'm not sure the previous code, but the newly added code is intentional. 
If my understanding is correct, `ActiveRecord::Migration::CheckPending` middleware does not support multi-db because it uses `ActiveRecord::Base.connection` in the first place. 
https://github.com/rails/rails/blob/400b2103540e1dfb370eab29a891687d778ad357/activerecord/lib/active_record/migration.rb#L562

If will support multi-db apps, may need to fix `ActiveRecord::Migration::CheckPending` middleware, but can we get the connection correctly at that time?